MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/offline/CacheManagementActivity.java

```diff
@@ -0,0 +1,284 @@
+   */
+  private void setNewMaxAmbientCache(long newMaxAmbientCacheSize) {
+    OfflineManager fileSource = OfflineManager.getInstance(CacheManagementActivity.this);
+    if (newMaxAmbientCacheSize < 0) {
+      showToast(R.string.invalid_max_ambient_cache_size);
+      return;
+    }
+
+    fileSource.setMaximumAmbientCacheSize(newMaxAmbientCacheSize,
+        new OfflineManager.FileSourceCallback() {
+          @Override
+          public void onSuccess() {
+            Toast.makeText(CacheManagementActivity.this, String.format(
+                getString(R.string.set_max_ambient_cache_size_toast_confirmation),
+                newMaxAmbientCacheSize), Toast.LENGTH_SHORT).show();
+          }
+
+          @Override
+          public void onError(@NonNull String message) {
+            showToast(R.string.set_max_ambient_cache_size_toast_error);
+          }
+        });
+  }
+
+  /**
+   * Clear the ambient cache. See ANDROID-DOCS-URL-HERE for more
+   * information about this method.
+   */
+  private void clearAmbientCache() {
+    OfflineManager fileSource = OfflineManager.getInstance(CacheManagementActivity.this);
+    fileSource.clearAmbientCache(new OfflineManager.FileSourceCallback() {
+      @Override
+      public void onSuccess() {
+        showToast(R.string.clear_ambient_cache_size_toast_confirmation);
+      }
+
+      @Override
+      public void onError(@NonNull String message) {
+        showToast(R.string.clear_ambient_cache_size_toast_error);
+      }
+    });
+  }
+
+  /**
+   * Reset the Maps SDK's database. See ANDROID-DOCS-URL-HERE for more
+   * information about this method.
+   */
+  private void resetDatabase() {
+    OfflineManager fileSource = OfflineManager.getInstance(CacheManagementActivity.this);
+    fileSource.resetDatabase(new OfflineManager.FileSourceCallback() {
+      @Override
+      public void onSuccess() {
+        showToast(R.string.reset_database_toast_confirmation);
+      }
+
+      @Override
+      public void onError(@NonNull String message) {
+        showToast(R.string.reset_database_toast_error);
+      }
+    });
+  }
+
+  /**
+   * Use Remote Config boolean values to make appropriate adjustments to the Maps SDK cache.
+   */
+  private void adjustCacheBasedOnRemoteConfigServerSettings() {
+    if (firebaseRemoteConfig.getBoolean(INVALIDATE_AMBIENT_CACHE_KEY)) {
+      invalidateAmbientCache();
+    }
+    if (firebaseRemoteConfig.getBoolean(SET_MAX_AMBIENT_CACHE_SIZE_BOOLEAN_KEY)) {
+      setNewMaxAmbientCache(firebaseRemoteConfig.getLong(SET_MAX_AMBIENT_CACHE_SIZE_AMOUNT_KEY));
+    }
+    if (firebaseRemoteConfig.getBoolean(CLEAR_AMBIENT_CACHE_KEY)) {
+      clearAmbientCache();
+    }
+    if (firebaseRemoteConfig.getBoolean(RESET_DATABASE_REMOTE_CONFIG_KEY)) {
+      resetDatabase();
+    }
+  }
+
+  private void showToast(int toastStringId) {
+    // validate that all callbacks occur on main thread
+    if (Looper.myLooper() == Looper.getMainLooper()) {
```

Comment:
This will not show a toast if this method is invoked from a worker thread. Could you try additionally post to the main thread?

Reply:
would the following work?

```java
if (Looper.myLooper() == Looper.getMainLooper()) {
      Toast.makeText(CacheManagementActivity.this,
          textToShow, Toast.LENGTH_SHORT).show();
    } else {
      CacheManagementActivity.this.runOnUiThread(new Runnable() {
        public void run() {
          Toast.makeText(CacheManagementActivity.this,
              textToShow, Toast.LENGTH_SHORT).show();
        }
      });
}
```

Reply:
`Activity#runOnUiThread` runs the code immediately if you're on UiThread already (basically performs the same check you did), so just
```
      CacheManagementActivity.this.runOnUiThread(new Runnable() {
        public void run() {
          Toast.makeText(CacheManagementActivity.this,
              textToShow, Toast.LENGTH_SHORT).show();
        }
      });
```
is enough.

Reply:
Ok, fixed.